### DISCUSSION TITLE
please help

### DISCUSSION THREAD

**Discussion by RustyShackIeford** [MLB-LED-Scoreboard/mlb-led-scoreboard#534] [General]
Hi i am very new to all of this, trying to get this to run and I am getting:

python3 main.py
Traceback (most recent call last):
  File "/home/pi/mlb-led-scoreboard/main.py", line 4, in <module>
    from data.screens import ScreenType
  File "/home/pi/mlb-led-scoreboard/data/__init__.py", line 5, in <module>
    from data import status
  File "/home/pi/mlb-led-scoreboard/data/status.py", line 1, in <module>
    from data.scoreboard.inning import Inning
  File "/home/pi/mlb-led-scoreboard/data/scoreboard/__init__.py", line 1, in <module>
    from data.game import Game
  File "/home/pi/mlb-led-scoreboard/data/game.py", line 5, in <module>
    import statsapi
ModuleNotFoundError: No module named 'statsapi'

any help is appreciated thanks

**Comment by behns13** [comment]
Same thing here.  Not sure why?  

I'm putting together a new scoreboard and ran into the same error.  I created this project a while ago on a separate pi and don't remember this occurring.

Anyone have any ideas?

**Reply by pjockey** [reply]
iIs the package/module `statsapi` installed? `pip3 list`. May need to start the venv and run that.

**Reply by behns13** [reply]
Doesn't look like statsapi was installed during the installation.  I wonder why.

**Reply by pjockey** [reply]
re-run the `install.sh`?

**Reply by behns13** [reply]
Just reran sudo ./install.sh

No dice :/ Could it be something with the sound device?

**Reply by pjockey** [reply]
```
pi@scoreboard:~/mlb-led-scoreboard $ source venv/bin/activate
(venv) pi@scoreboard:~/mlb-led-scoreboard $ pip3 list
Package               Version
--------------------- -----------
bdfparser             2.2.0
certifi               2024.2.2
charset-normalizer    3.3.2
feedparser            6.0.10
geojson               2.5.0
idna                  3.6
libsixel-python       0.5.0
**MLB-StatsAPI          1.7.1**
numpy                 1.26.4
pillow                10.2.0
pip                   20.3.4
pkg-resources         0.0.0
pygame                2.5.2
pyowm                 3.3.0
PySocks               1.7.1
pytz-deprecation-shim 0.1.0.post0
requests              2.31.0
rgbmatrix             0.0.1
RGBMatrixEmulator     0.11.3
setuptools            44.1.1
sgmllib3k             1.0.0
tornado               6.4
tzdata                2024.1
tzlocal               4.2
urllib3               2.2.1
```

**Reply by behns13** [reply]
Oh, using venv, I do see the MLB StatsAPI now.

**Reply by pjockey** [reply]
I don't quite understand the getting into the venv by starting the `main.py `fuke which is outside of the venv.

**Reply by pjockey** [reply]
I have have the MLB_StatsAPI also installed in the global evironment, that's perhaps why mine is actually functioning.

**Comment by pjockey** [comment]
sudo python3 main.py --led-gpio-mapping=adafruit-hat-pwm --led-rows=64 --led-cols=128 --led-rgb-sequence=BGR --led-brightness=60  --led-slowdown-gpio=2 --led-row-addr-type=3

This is my starting lines.

**Reply by behns13** [reply]
I might be wrong, but I don't believe that the flags have an impact here.

It feels like an installation step the gets skipped for some reason.

**Reply by pjockey** [reply]
package name is MLB-StatsAPI

**Comment by ty-porter** [comment]
Friendly reminder -- v8 and higher changes the startup commands.

```sh
sudo ./main.py <FLAGS>
```

https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard?tab=readme-ov-file#usage

**Reply by behns13** [reply]
Ope.

That was it.  Appreciate it!

**Comment by RustyShackIeford** [comment]
I’m still having this issue. 

When trying to manually install statsapi I get this:

DEPRECATION: Loading egg at /home/pi/nhlsb-venv/lib/python3.11/site-packages/rgbmatrix-0.0.1-py3.11-linux-armv7l.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
ERROR: Could not find a version that satisfies the requirement statsapi (from versions: none)
ERROR: No matching distribution found for statsapi

I am also running the nhl led board separately 

I can see the mlb logo flashing on my matrix panel but that’s about it seems like a boot loop

**Reply by pjockey** [reply]
Did you try the `sudo ./main.py `command as noted above? 

https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard?tab=readme-ov-file#usage

**Reply by RustyShackIeford** [reply]
yeah! Double checked and it appeared I had an error in my config file,

working great now! 

Thanks everyone for the assistance!
